src/effects/effectslot.cpp

```diff
@@ -299,15 +299,15 @@ EffectButtonParameterSlotPointer EffectSlot::getEffectButtonParameterSlot(unsign
     return m_buttonParameters[slotNumber];
 }
 
-bool EffectSlot::loadEffect(EffectManifestPointer pManifest, EffectInstantiatorPointer pInstantiator,
+void EffectSlot::loadEffect(EffectManifestPointer pManifest, EffectInstantiatorPointer pInstantiator,
         const QSet<ChannelHandleAndGroup>& activeChannels) {
     // Remove an effect if already loaded in slot
     if (m_pEngineEffect != nullptr) {
```

Comment:
remove this `if`. `unloadEffect()` is needed in every case.

Reply:
I did not remove this to maintain a sanity check of calling unloadEffect only if the effectslot is already loaded. An assertion is present in removeFromEngine method for the same.

Reply:
Oh sorry, you are right.